doc/tutorial.md

```diff
@@ -1445,6 +1445,48 @@ A `pipe<T>` resembles a Unix pipe, in that it has a read side, a write side, and
 The pipe's read and write interfaces are future-based blocking. I.e., the write() and read() methods return a future which is fulfilled when the operation is complete. The pipe is single-reader single-writer, meaning that until the future returned by read() is fulfilled, read() must not be called again (and same for write).
 Note: The pipe reader and writer are movable, but *not* copyable. It is often convenient to wrap each end in a shared pointer, so it can be copied (e.g., used in an std::function which needs to be copyable) or easily captured into multiple continuations.
 
+The following example demonstrates a scenario in which two different continuations communicate with each other by using a pipe.
+```cpp
+#include <seastar/core/future.hh>
+#include <seastar/core/do_with.hh>
+#include <seastar/core/pipe.hh>
+#include <seastar/core/when_all.hh>
+#include <iostream>
+
+
+seastar::future<> f() {
+    return seastar::do_with(seastar::pipe<int>(1), [] (auto& pipe) {
+        auto reader = [&pipe]()->seastar::future<> {
+            std::cout << "[Reader] Waiting for data..." << std::endl;
+            return pipe.reader.read().then([&pipe] (auto data){
+                std::cout << "[Reader] In the first read we got " << data.value() << std::endl;
+
+                return pipe.reader.read();
+            }).then([] (auto data) {
+                std::cout << "[Reader] In the second read we got " << data.value() << std::endl;
+            });
+        };
+
+        auto writer = [&pipe]()->seastar::future<> {
+            std::cout << "[Writer] Writing data to the queue..." << std::endl;
+            return pipe.writer.write(1).then([&pipe] {
+                std::cout << "[Writer] We finished write one" << std::endl;
+
+                return pipe.writer.write(2);
+            }).then([] {
+                std::cout << "[Writer] We finished write two" << std::endl;
+            });
+        };
+
+        return seastar::when_all(std::move(reader), std::move(writer)).discard_result();
+    });
+}
+```
+
+In this example we have a function `reader` which gets a pipe and waits for data. We have a second function `writer` which sends data.
+We define a queue with buffer size 1. Thus, the writer can write a single object, int in our case, and wait for a read event before the next write.
+The reader blocks until there is data in the pipe and prints the value it got.
+
```

Comment:
I'd like to discourage continuation style in favor of coroutines, which are more readable (and safer). Care to adjust the example?